### DISCUSSION TITLE
Custom filters mode

### DISCUSSION THREAD

**Discussion by Gorbiel** [ThioJoe/YT-Spammer-Purge#365] [Ideas]
First of all, I love the project just as much as I hate the spam bots and I want to thank you for creating it.

You should add the option to create custom filters with a lot of different rules like MAX_EMOJIS_USED that will flag every comment that has more emojis than specified (it doesn't have to be emojis but also special symbols and non-ascii characters)

even simple word-based filter (BANNED_WORDS) would help - for example a certain popular youtuber/streamer could ban the use of the word "simp" but I won't throw any specific names
you could always make an advanced word filter that will also look for different ways that people tend to go around to avoid detection

you could add the option REMOVE_DUPLICATES, because some people write the exact same comment more than once and in case it's not a perfect match it could leave only the newest comment

Should anyone have more suggestions for filters, please join the discussion

**Comment by ThioJoe** [comment]
Yea something like this wouldn't be a bad idea. I will point out there is already the ability to search for any word/sentence

**Reply by Gorbiel** [reply]
That's already a lot, but of course some of these filters would need to be limited to your own channel only, because people might use it in malicious ways and report innocent users (it's the internet after all)

**Reply by TechStudent10** [reply]
Or maybe you could create an entire ecosystem of filters. Kind of like how Python has libraries. People can make them public, and users can download and import them as a filter. You could have an entire filter library!

**Comment by Rairye** [comment]
@Gorbiel What kinds of ways of going around detection are you thinking of? There are solutions for common methods that people might use to try to bypass detection.

Does the program already have a method for detecting emojis/special characters in strings? There is a simple, single-method Python solution for detecting emojis/special characters in a way that generates few false positives.
